### DISCUSSION TITLE
How to querying a object from object data in prisma with N Relations

### DISCUSSION THREAD

**Discussion by rahXephonz** [prisma/prisma#12302] [Q&A] [answered]
Hello Everyone, Hope you are good
 
Get some problem again, and i want to ask why my result after find a many comments in comments got this result

![image](https://user-images.githubusercontent.com/77242429/158051642-4e147290-8438-4b63-9d78-7364abeacf44.png)

How to get a comments object real like 
```json
{
  id: 1,
  ...blablabla
} 
```

Not [Object] See at the photos

my code
```ts
  getCommentsFromComents: async (
    _: any,
    // this is args from query graphql
    { commentId }: OpArgs,
    { db }: Context
  ) => {
    const allComments = await db.comment.findMany({
      where: {
        id: commentId,
      },

      include: {
        user: true,
        comments: {
          include: { user: true },
        },
      },
    });

    console.log(allComments);

    return {
      comments: allComments,
      // cursor: comments.length ? comments[comments.length - 1].id : null,
      // hasMore: comments.length
      //   ? comments[comments.length - 1].id !==
      //     allComments[allComments.length - 1].id
      //   : false,
    };
  },
  ```
  
  my prisma model
  ```prisma
  model Comment {
  id        String   @id @default(cuid())
  text      String
  post      Post     @relation(fields: [postId], references: [id])
  user      User     @relation(fields: [userId], references: [id])
  userId    String
  postId    String
  comments  Comment[] @relation("CommentToComment")
  Comment   Comment?  @relation("CommentToComment", fields: [commentId], references: [id])
  commentId String? 
  date      DateTime @default(now())


  @@map(name: "comments")
}
```

```gql
// schema
type Query {
    getCommentsFromComents(
      commentId: ID!
      pageSize: Int
      after: String
    ): CommentsConnection!
}

type CommentsConnection {
    count: Int
    cursor: String
    hasMore: Boolean!
    comments: [Comment]!
}

type Comment {
    id: ID!
    text: String!
    user: User!
    post: Post!
    date: DateTime
    comments: [Comment]
}
```

In here in my model prisma that have comments and every comments including user relations
and i want querying it like : 
```gql

query comments {
  id
  text
  user {
    id
    name
   }
  comments{
    id
    text
    user {
     id
     name
     }
   }
 }
 ```

**Comment by austincrim** [comment, marked as answer]
Hey @ioofy 👋,

Everything here looks correct to me! `comments` is just being represented as `Object` in the console output. You could use `JSON.stringify` to see the full output in your `console.log`

```ts
console.log(JSON.stringify(allComments, null, 2))
```

**Reply by rahXephonz** [reply]
it was solved two days ago, thanks again for replying my answer sir :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
